core/commands/repo.go

```diff
@@ -87,26 +110,29 @@ order to reclaim hard disk space.
 				return nil, err
 			}
 
-			marshal := func(v interface{}) (io.Reader, error) {
-				obj, ok := v.(*corerepo.KeyRemoved)
+			for v := range outChan {
```

Comment:
@whyrusleeping for example here is the output from a test case when run with `-v`
```
expecting success: 
                test_must_fail ipfs repo gc --stream-errors 2>&1 | tee repo_gc_out &&
                grep -q "Error: could not retrieve links for $LEAF1" repo_gc_out &&
                grep -q "Error: could not retrieve links for $LEAF2" repo_gc_out &&
                grep -q "Error: garbage collection aborted" repo_gc_out

Error: could not retrieve links for QmSijovevteoY63Uj1uC5b8pkpDU5Jgyk2dYBqz3sMJUPc: merkledag: not found
Error: could not retrieve links for QmTbPEyrA1JyGUHFvmtx1FNZVzdBreMv8Hc8jV9sBRWhNA: The block referred to by 'QmTbPEyrA1JyGUHFvmtx1FNZVzdBreMv8Hc8jV9sBRWhNA' was not a valid merkledag node
Error: garbage collection aborted: could not retrieve some links
01:35:09.788 ERROR commands/h: encountered errors during gc run client.go:247
Error: encountered errors during gc run
ok 48 - 'ipfs repo gc --stream-errors' should abort and report each error separately

```